### PR TITLE
GitHub Actions migration

### PR DESCRIPTION
**Pull request by Tbaile**
Moving all the CI scripts from Travis to GitHub, to edit the retrocompatibility for Travis here's the list of the changes:
- Moved all \*-travis scripts to \*-ci and renamed the Travis-dependent variables
- Moved all the CI provider variables definition to \*-travis and \*-github

Due to GitHub limitation, there's no support for pushing to `packages.nethserver.org` inside fork environments. A workaround will be applied if needed, but no intention is to support that as for now.

NethServer/dev#6700

### CODE REVIEW

File: buildimage/uploadrpms-ci [hunk 1]
```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+#
+# Copyright (C) 2017 Nethesis S.r.l.
```
Comment (gsanchietti) [inline buildimage/uploadrpms-ci:4]:
```suggestion
# Copyright (C) 2022 Nethesis S.r.l.
```

File: README.rst [hunk 2]
```diff
@@ -211,75 +211,92 @@ To make ``--follow-tags`` permanent run this command: ::
...
 
-To automate the RPM build process using Travis CI
+To automate the RPM build process using GitHub
 
-* create a ``.travis.yml`` file inside the source code repository hosted on
+* create a ``.github/workflows/make-rpms.yml`` file inside the source code repository hosted on
   GitHub.
 
-* the repository must have Travis CI builds enabled and upload secrets properly set up.
+* the repository must have builds enabled and upload secrets properly set up, if the repo is inside Nethserver org, everything is already set up (as long as you have write access to the repo).
```
Comment (Amygos) [inline README.rst:229]: we can also add that the organization admin must allow access to relative the organization's secrets from the package repository.

File: README.rst [hunk 3]
```diff
@@ -211,75 +211,92 @@ To make ``--follow-tags`` permanent run this command: ::
...
+        NSVER: 7
+        DOCKER_IMAGE: nethserver/makerpms:7
+      steps:
+        - uses: actions/checkout@v3
+          with:
+            fetch-depth: 0
+            ref: ${{ github.head_ref }}
+        - name: Generate .env file
+          run: |
+            cat .env < EOF
```
Comment (DavidePrincipi) [inline README.rst:255]: Fix redirections syntax

```suggestion
            cat > .env <<EOF
```

File: buildimage/uploadrpms [hunk 4]
```diff
@@ -37,12 +37,21 @@ tmpcmd=$(mktemp)
...
     echo "mkdir $spool";
     echo "cd $spool";
-    for F in $(find . -name '*.rpm'); do
-        echo "put $F";
-    done;
+    if [[ "$GITHUB_ACTIONS" && "$GPG_SIGN_KEY" ]]; then
+        echo "$GPG_SIGN_KEY" | gpg --import &> /dev/null
+        while IFS= read -r -d '' file
+        do
+            gpg --encrypt --yes --trust-model always --user nethinfra@nethesis.it "$file" &> /dev/null
```
Comment (DavidePrincipi) [inline buildimage/uploadrpms:47]: Stderr might contain useful information to understand and debug this. I'd send it to the default descriptor (build log).
```suggestion
        echo "$GPG_SIGN_KEY" | gpg --import > /dev/null
        while IFS= read -r -d '' file
        do
            gpg --encrypt --yes --trust-model always --user nethinfra@nethesis.it "$file" > /dev/null
```

**Review by DavidePrincipi — APPROVED** [review]

**Review by gsanchietti — COMMENTED** [review]
If for @Amygos and @DavidePrincipi is ok, let's go with the merge ;)

**Comment by Tbaile** [comment]
I've updated the branch removing the GPG check, turns out that you need a lot of effort to manage correctly GPG keys in env (especially inside GH Actions). Other commits are to allow server scripts to fetch correctly logs, allowing check on uploaded rpms.

File: README.rst [hunk 5]
```diff
@@ -211,75 +211,92 @@ To make ``--follow-tags`` permanent run this command: ::
...
+                nethserver/build \
+                uploadrpms-github
+              echo "Publish complete."
+            fi
+            rm .env
 
 Usage
 ^^^^^
 
 Travis CI builds are triggered automatically when:
```
Comment (DavidePrincipi) [inline README.rst:296]: It seems there are still some `travis` to search and replace in this file, like this one.

File: README.rst [hunk 6]
```diff
@@ -359,6 +376,6 @@ If ``DEST_ID=forge``:
 .. rubric:: References
 
 .. [#Podman] Podman is a daemonless Linux container engine. https://podman.io/
-.. [#Autobuild] Is a particular kind of repository in ``packages.nethserver.org`` that hosts the rpms builded automatically from travis-ci.com. http://packages.nethserver.org/nethserver/7.4.1708/autobuild/x86_64/Packages/
-.. [#Testing] Is a repository in ``packages.nethserver.org`` that hosts the rpms builded automatically from travis-ci.com started form official ``nethserver`` github repository. http://packages.nethserver.org/nethserver/7.4.1708/testing/x86_64/Packages/
+.. [#Autobuild] Is a particular kind of repository in ``packages.nethserver.org`` that hosts the rpms builded automatically from github.com/features/actions. http://packages.nethserver.org/nethserver/7.4.1708/autobuild/x86_64/Packages/
+.. [#Testing] Is a repository in ``packages.nethserver.org`` that hosts the rpms builded automatically from github.com/features/actions started form official ``nethserver`` github repository. http://packages.nethserver.org/nethserver/7.4.1708/testing/x86_64/Packages/
```
Comment (Amygos) [inline README.rst:380]:
```suggestion
.. [#Autobuild] Is a particular kind of repository in ``packages.nethserver.org`` that hosts the rpms builded automatically from github.com/features/actions. http://packages.nethserver.org/nethserver/7.9.2009/autobuild/x86_64/Packages/
.. [#Testing] Is a repository in ``packages.nethserver.org`` that hosts the rpms builded automatically from github.com/features/actions started form official ``nethserver`` github repository. http://packages.nethserver.org/nethserver/7.9.2009/testing/x86_64/Packages/
```

**Review by Amygos — APPROVED** [review]

**Comment by Tbaile** [comment]
Check fails due to missing secrets for pushing image to registry, shall I merge anyway? (The testing has been already done upstream with [NethServer/nethserver-dummy-rpm](https://github.com/NethServer/nethserver-dummy-rpm) and a custom registry entry pushed by me)

**Pull request merged**